dotnet/src/SemanticKernel.UnitTests/Configuration/KernelConfigTests.cs

```diff
@@ -170,227 +165,219 @@ public void ItCanOverwriteBackends()
+        target.RemoveTextCompletionService("1");
+        Assert.Equal("2", target.DefaultTextCompletionServiceId);
+        target.RemoveTextCompletionService("2");
+        Assert.Equal("3", target.DefaultTextCompletionServiceId);
+        target.RemoveTextCompletionService("3");
+        Assert.Null(target.DefaultTextCompletionServiceId);
     }
 
     [Fact]
     public void ItCanRemoveOneEmbeddingsBackend()
     {
         // Arrange
         var target = new KernelConfig();
-        target.AddAzureOpenAIEmbeddingsBackend("1", "dep", "https://localhost", "key");
-        target.AddAzureOpenAIEmbeddingsBackend("2", "dep", "https://localhost", "key");
-        target.AddOpenAIEmbeddingsBackend("3", "model", "key");
-        Assert.Equal("1", target.DefaultEmbeddingsBackend);
+        target.AddAzureOpenAIEmbeddingGeneration("1", "dep", "https://localhost", "key");
+        target.AddAzureOpenAIEmbeddingGeneration("2", "dep", "https://localhost", "key");
+        target.AddOpenAIEmbeddingGeneration("3", "model", "key");
+        Assert.Equal("1", target.DefaultTextEmbeddingServiceId);
 
         // Act - Assert
-        target.RemoveEmbeddingsBackend("1");
-        Assert.Equal("2", target.DefaultEmbeddingsBackend);
-        target.RemoveEmbeddingsBackend("2");
-        Assert.Equal("3", target.DefaultEmbeddingsBackend);
-        target.RemoveEmbeddingsBackend("3");
-        Assert.Null(target.DefaultEmbeddingsBackend);
+        target.RemoveTextEmbeddingService("1");
+        Assert.Equal("2", target.DefaultTextEmbeddingServiceId);
+        target.RemoveTextEmbeddingService("2");
+        Assert.Equal("3", target.DefaultTextEmbeddingServiceId);
+        target.RemoveTextEmbeddingService("3");
+        Assert.Null(target.DefaultTextEmbeddingServiceId);
     }
 
     [Fact]
     public void GetEmbeddingsBackendItReturnsDefaultWhenNonExistingLabelIsProvided()
     {
         // Arrange
         var target = new KernelConfig();
-        var defaultBackendLabel = "2";
-        target.AddAzureOpenAIEmbeddingsBackend("1", "dep", "https://localhost", "key");
-        target.AddAzureOpenAIEmbeddingsBackend(defaultBackendLabel, "dep", "https://localhost", "key");
-        target.SetDefaultEmbeddingsBackend(defaultBackendLabel);
+        target.AddOpenAIEmbeddingGeneration("1", "dep", "https://localhost", "key");
+        target.AddAzureOpenAIEmbeddingGeneration("2", "dep", "https://localhost", "key");
+        target.SetDefaultEmbeddingService("2");
 
         // Act
-        var result = target.GetEmbeddingsBackend("test");
+        var result = target.GetTextEmbeddingServiceIdOrDefault("test");
 
         // Assert
-        Assert.IsType<AzureOpenAIConfig>(result);
-        Assert.Equal(defaultBackendLabel, result.Label);
+        Assert.Equal("2", result);
     }
 
     [Fact]
-    public void GetEmbeddingsBackendItReturnsSpecificWhenExistingLabelIsProvided()
+    public void GetEmbeddingsBackendReturnsSpecificWhenExistingLabelIsProvided()
```

Comment:
We will also need to change all `Backends` occurrences in this file to `Services`.